src/skin/legacyskinparser.cpp

```diff
@@ -304,19 +306,34 @@ QWidget* LegacySkinParser::parseSkin(QString skinPath, QWidget* pParent) {
 
     SkinManifest manifest = getSkinManifest(skinDocument);
 
-    // Apply SkinManifest attributes.
+    // Apply SkinManifest attributes by looping through the proto.
     for (int i = 0; i < manifest.attribute_size(); ++i) {
         const SkinManifest::Attribute& attribute = manifest.attribute(i);
         if (!attribute.has_config_key()) {
             continue;
         }
-        ConfigKey configKey = ConfigKey::parseCommaSeparated(
-            QString::fromStdString(attribute.config_key()));
 
         bool ok = false;
         double value = QString::fromStdString(attribute.value()).toDouble(&ok);
         if (ok) {
-            ControlObject::set(configKey, value);
+            ConfigKey configKey = ConfigKey::parseCommaSeparated(
+                    QString::fromStdString(attribute.config_key()));
+            // Set the specified attribute, possibly creating the control
+            // object in the process.
+            bool created = false;
+            // Update the value specified in the skin with whatever is in the
+            // saved config, if any.
+            value = m_pConfig->getValueString(configKey,
+                                              QString::number(value)).toDouble();
+            ControlObject* pControl = controlFromConfigKey(configKey,
+                                                           attribute.persist(),
+                                                           &created);
+            pControl->set(value);
```

Comment:
We have similar code in ControlDoublePrivate::initialize()
There we have the hard coded default = 0. Please pipe the initial persistent value down to this function.  
We may also consider to adopt it as the "default" value  ControlDoublePrivate::m_defaultValue